core/domain/takeout_service.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Functions to export the data of all user related models from a given
+user_id.
+"""
+from __future__ import absolute_import   # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+
+(
+    user_models, collection_models, exploration_models, feedback_models,
+    suggestion_models, email_models) = models.Registry.import_models([
+        models.NAMES.user, models.NAMES.collection, models.NAMES.exploration,
+        models.NAMES.feedback, models.NAMES.suggestion, models.NAMES.email])
+
+
+def export_data_for_user(user_id):
+    """Exports selected models according to model defined export_data functions.
+
+    Args:
+            user_id: str. The user_id of the user to export.
+
+    Returns:
+            dict. Dictionary containing all user data.
+    """
+
+    models_export_format = {
+        'stats_data': user_models.UserStatsModel,
```

Comment:
@varun-tandon, This looks good to me butis there any way we can make this name generatable out of the model name? I think we follow such a pattern in `prod_validation_jobs_one_off`.
@ankita240796, do you have any suggestions on that?

(I'm fine with this way as well.)

Reply:
Hi @DubeySandeep, I checked this but we do not have any specific method which can form names based on model names. We supply a dict of id names with the model class name ([Ref](https://github.com/oppia/oppia/blob/develop/core/domain/prod_validation_jobs_one_off.py#L4652)). 

We can try generating these names out of model names though if they exactly match the model name in lower case separated by underscore. Something of this pattern is done [here](https://github.com/oppia/oppia/blob/develop/core/domain/prod_validation_jobs_one_off.py#L412).

Thanks!

Reply:
@DubeySandeep i'm marking as resolved because I don't think this works in our use case. Feel free to correct me/unresolve if I am wrong

Reply:
I'm fine if that's not possible but can you please leave the reason behind "Why it's not possible" in this thread?

Reply:
Hi @varun-tandon, I think you can generate names using Model class names and it would be a good idea to do this to maintain consistency.

For example, If name of a model is `XabcYabcZabcModel`, then the generated name should be `x_y_z_data`. You can split the model name by uppercase letters and then select each uppercase letter, convert it lowercase and join them by underscores. Further append `_data` in the end.

Do let me know if anything is not clear and you need any code snippet for the same from my side.

Thanks!

Reply:
Hi @ankita240796 I'm a bit confused how I would take the model instance and get its name. I don't quite understand the snippet you sent above. Thanks for the help!

Reply:
@ankita240796, Can you please reply to this thread?

@varun-tandon, In general, if you want someone to reply or review your PR, feel free to add then in the assignee list.

Reply:
Hi @varun-tandon, you can get the name of model instance from the class name as follows:


```
class ModelName:
   Model Code
```

```
import re

name = ModelName.__name__ 
>>> name = 'ModelName'

split_name = re.findall('[A-Z][^A-Z]*', name)
>>> split_name = ['Model',  'Name']

final_name = ('_').join([item.lower() for item in split_name])
>>> final_name = 'model_name'

final_name = final_name + '_data'
>>> final_name = 'model_name_data'
```

Please do let me know if anything is not clear and I will further explain it to you (Assign me to the PR as mentioned by @DubeySandeep if you have other queries).

Thanks!

Reply:
Got it, thanks!